### PR TITLE
Reworked interest system branch

### PR DESCRIPTION
**Pull request by oldmud0**
This is a reworked version of the mess that was made trying to make an interest system, after an interactive rebase with #18 and #19.

The feature is far from finished; please work from this branch and be sure to set your editor to two spaces per tab.

### CODE REVIEW

File: src/codeu/chat/common/BasicController.java [hunk 1]
```diff
@@ -51,4 +51,9 @@
   //  Whether conversations can have the same title is undefined.
   ConversationHeader newConversation(String title, Uuid owner);
 
+  String newUserInterest(String name, Uuid signedInId);
```
Comment (cbick) [inline src/codeu/chat/common/BasicController.java:54]: some comments on these (doesn't need too much, as these are fairly self-explanatory)

File: src/codeu/chat/common/BasicView.java [hunk 2]
```diff
@@ -55,10 +55,12 @@
...
   // GET VERSION
   //
-  //   Return the server's version.
   //   Return the server's version, or a null UUID if unavailable.
   VersionInfo getVersion();
+
   ServerInfo getInfo();
-  String getAllConvosFromServer();
+
+  String getAllConvosFromServer(Uuid signedInId);
```
Comment (cbick) [inline src/codeu/chat/common/BasicView.java:63]: comments here too

File: src/codeu/chat/common/RawController.java [hunk 3]
```diff
@@ -42,4 +42,8 @@
   // already in use, the call will fail and null will be returned.
   ConversationHeader newConversation(Uuid id, String title, Uuid owner, Time creationTime);
 
+  String newUserInterest(String name, Uuid signedInId);
```
Comment (cbick) [inline src/codeu/chat/common/RawController.java:45]: comments

File: src/codeu/chat/common/VersionInfo.java [hunk 4]
```diff
@@ -44,8 +44,11 @@ public VersionInfo(Uuid version) {
   public Uuid getVersion() {
     return version;
   }
-  
-  //Retrieves the stored UUID form of the version.
```
Comment (cbick) [inline src/codeu/chat/common/VersionInfo.java:48]: was it intentional to delete this comment?

File: src/codeu/chat/server/Controller.java [hunk 5]
```diff
@@ -128,6 +128,30 @@ public User newUser(Uuid id, String name, Time creationTime) {
   }
 
   @Override
+  public String newUserInterest(String name, Uuid signedInId) {
```
Comment (cbick) [inline src/codeu/chat/server/Controller.java:131]: these pieces seem non-functional?

File: src/codeu/chat/server/Server.java [hunk 6]
```diff
@@ -18,419 +18,495 @@
...
+        final Uuid author = Uuid.SERIALIZER.read(in);
+        User signedInUser = view.findUser(author);
+        final Uuid conversation = Uuid.SERIALIZER.read(in);
+        final String content = Serializers.STRING.read(in);
+        ConversationHeader convo = view.findConversation(conversation);
+        // for user status update
+        userInterests.get(signedInUser).addModifiedConversation(convo.title);
+        // for convo status update
+        for (User temp : userInterests.keySet()){
+          if (userInterests.get(temp).getInterestedConvos().isEmpty() == false){
```
Comment (cbick) [inline src/codeu/chat/server/Server.java:98]: this check seems redundant with the one nested within it

File: src/codeu/chat/server/Server.java [hunk 7]
```diff
@@ -18,419 +18,495 @@
...
+          }
+        }
+
+        Uuid interestID = interest.id;
+        final Uuid signedInId = Uuid.SERIALIZER.read(in);
+        User signedInUser = view.findUser(signedInId);
+        User temp = null;
+
+        for (User key : userInterests.keySet()) {
+          if(key.equals(signedInUser)){
```
Comment (cbick) [inline src/codeu/chat/server/Server.java:236]: can't we just say userInterests.get(signedInUser) ?

File: src/codeu/chat/server/Server.java [hunk 8]
```diff
@@ -18,419 +18,495 @@
...
+        for(User temp : userInterests.keySet()) {
+          if(temp.name.equals(interestName)) {
+            interest = temp;
+          }
+        }
+
+        Uuid interestID = interest.id;
+        final Uuid signedInId = Uuid.SERIALIZER.read(in);
+        User signedInUser = view.findUser(signedInId);
+        User temp = null;
```
Comment (cbick) [inline src/codeu/chat/server/Server.java:233]: this never seems to get used for any purpose

File: src/codeu/chat/server/Server.java [hunk 9]
```diff
@@ -18,419 +18,495 @@
...
+        final User signedInUser = view.findUser(signedInId);
+        User temp = null;
+
+        for (User key : userInterests.keySet()) {
+          if (key.equals(signedInUser)) {
+            temp = key;
+          }
+        }
+
+        final Set<Uuid> ids = userInterests.get(temp).getInterestedUserIds();
```
Comment (cbick) [inline src/codeu/chat/server/Server.java:261]: why not just userInterests.get(key) ?

File: src/codeu/chat/server/View.java [hunk 10]
```diff
@@ -131,4 +131,16 @@ public VersionInfo getVersion() {
   public ServerInfo getInfo() {
     return info;
   }
+
+  //TODO return a specific
```
Comment (cbick) [inline src/codeu/chat/server/View.java:135]: these seem unimplemented

**Review by cbick — CHANGES_REQUESTED** [review]

File: src/codeu/chat/common/VersionInfo.java (hunk 4 above)
Comment (oldmud0) [reply src/codeu/chat/common/VersionInfo.java:48]: It's already in line 42.

File: src/codeu/chat/util/ServerInfo.java [hunk 11]
```diff
@@ -9,8 +9,6 @@ public ServerInfo(Time startTime) {
 		this.startTime = startTime;
 	}
 	public String toString(){
-		String out = "";
-		out = startTime.toString();
-		return out;
```
Comment (oldmud0) [inline src/codeu/chat/util/ServerInfo.java:14]: I do believe I had to manually remove this from the merge. I really don't know why this simple operation would be done in such a protracted manner.

**Pull request merged**